Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -1266,13 +1266,13 @@ - (void)takeSnapshotWithConfiguration:(WKSnapshotConfiguration *)snapshotConfigu
         return;
     }
 
-    _page->callAfterNextPresentationUpdate([callSnapshotRect = WTFMove(callSnapshotRect), handler](WebKit::CallbackBase::Error error) mutable {
-        if (error != WebKit::CallbackBase::Error::None) {
-            tracePoint(TakeSnapshotEnd, snapshotFailedTraceValue);
-            handler(nil, createNSError(WKErrorUnknown).get());
-            return;
-        }
+    if (!_page->hasRunningProcess() || !_page->drawingArea()) {
+        tracePoint(TakeSnapshotEnd, snapshotFailedTraceValue);
+        handler(nil, createNSError(WKErrorUnknown).get());
+        return;
+    }
 
+    _page->callAfterNextPresentationUpdate([callSnapshotRect = WTFMove(callSnapshotRect), handler] () mutable {
```

Comment:
Since you have removed the error param, are these still guaranteed to happen, like before? How do you know if it succeeded or failed? If you've made them *not* guaranteed to happen (queued up on the UI side and flushed on crash/etc.), that is not OK.

Reply:
I'll add another check inside the completion handler which should match existing behavior if a process crashes during callAfterNextPresentationUpdate.  The CompletionHandlers are all called when a Connection closes after a crash.

Reply:
OK!